GVRf/Framework/src/org/gearvrf/jassimp2/GVRJassimpSceneObject.java

```diff
@@ -14,15 +16,30 @@
 import org.gearvrf.GVRTexture;
 import org.gearvrf.GVRMaterial.GVRShaderType;
 import org.gearvrf.utility.Log;
+import org.gearvrf.animation.keyframe.GVRKeyFrameAnimation;
 
 public class GVRJassimpSceneObject extends GVRSceneObject {
     private static final String TAG = GVRJassimpSceneObject.class.getSimpleName();
     protected AiScene scene;
+    protected List<GVRKeyFrameAnimation> animations;
 
     public GVRJassimpSceneObject(GVRContext gvrContext, AiScene scene) {
         super(gvrContext);
-        this.scene = scene;
-        recurseAssimpNodes(this, scene.getSceneRoot(GVRJassimpAdapter.sWrapperProvider));
+        animations = new ArrayList<GVRKeyFrameAnimation>();
+
+        if (scene != null) {
+            this.scene = scene;
+            recurseAssimpNodes(this, scene.getSceneRoot(GVRJassimpAdapter.sWrapperProvider));
+
+            // Animations
+            for (AiAnimation aiAnim : scene.getAnimations()) {
+                animations.add(GVRJassimpAdapter.get().createAnimation(aiAnim, this));
+            }
+        }
+    }
+
+    public List<GVRKeyFrameAnimation> getAnimations() {
```

Comment:
This method requires an explicit cast to GVRJassimpSceneObject. Isn't GVRJassimpSceneObject an implementation detail?